include/SDL3/SDL_gpu.h

```diff
@@ -24,7 +24,101 @@
+ * instructions that will be submitted to the GPU in batch. Complex scenes
+ * can use multiple command buffers, maybe configured across multiple threads
+ * in parallel, as long as they are submitted in the correct order, but many
+ * apps will just need one command buffer per frame.
+ *
+ * Rendering can happen to a texture (what other APIs call a "render target")
+ * or it can happen to the swapchain (which is just a special texture that
+ * represents a window's contents).
+ *
+ * Rendering actually happens in a Render Pass, which is encoded into a
+ * command buffer. One can encode multiple render passes (or alternate
+ * between render and compute passes) in a single command buffer, but many
+ * apps might simply need a single render pass in a single command buffer.
+ *
+ * The app calls SDL_BeginGPURenderPass(). If it isn't rendering to a texture,
+ * the app can use SDL_AcquireGPUSwapchainTexture() to render to the window.
+ * Then it sets states it needs for each draw:
+ *
+ * - SDL_BindGPUGraphicsPipeline
+ * - SDL_SetGPUViewport
+ * - SDL_BindGPUVertexBuffers
+ * - SDL_BindGPUVertexSamplers
+ * - etc
+ *
+ * Then, make the actual draw commands with these states:
+ *
+ * - SDL_DrawGPUPrimitives
+ * - SDL_DrawGPUPrimitivesIndirect
+ * - SDL_DrawGPUIndexedPrimitivesIndirect
+ * - etc
+ *
+ * It can then set new states and make new draws in the same command buffer,
+ * until the whole scene is rendered.
+ *
+ * After all the drawing commands are complete, the app should call
+ * SDL_EndGPURenderPass(), then SDL_SubmitGPUCommandBuffer() to send
+ * it to the GPU for processing.
+ *
+ * If the app needs to read back data from texture or buffers, the API
+ * has an efficient way of doing this, provided that the app is willing to tolerate some latency.
+ * When the app uses SDL_DownloadFromGPUTexture or SDL_DownloadFromGPUBuffer, submitting the command buffer with
+ * SubmitGPUCommandBufferAndAcquireFence() will return a fence handle that the app
+ * can poll or wait on in a thread. Once the fence indicates that the command buffer is done processing,
+ * it is safe to read the downloaded data. Make sure to call SDL_ReleaseGPUFence() when done with the fence.
+ *
+ * The API also has "compute" support. The app calls SDL_GPUBeginComputePass()
+ * with compute-writeable textures and/or buffers, which can be written to in a compute shader.
+ * Then it sets states it needs for the compute dispatches:
+ *
+ * - SDL_BindGPUComputePipeline
+ * - SDL_BindGPUComputeStorageBuffers
+ * - SDL_BindGPUComputeStorageTextures
+ *
+ * Then, dispatch compute work:
+ *
+ * - SDL_DispatchGPUCompute
+ *
+ * For advanced users, this opens up powerful GPU-driven workflows.
+ *
+ * This is an extremely quick overview that leaves out several important
+ * details. Already, though, one can see that GPU programming can be quite
+ * complex! If you just need simple 2D graphics, the
+ * [Render API](CategoryRender) is much easier to use but still
```

Comment:
Can you put a web link in here? This is in the header, which may be read off of the wiki.